pkg/ext-proc/main.go

```diff
@@ -144,53 +157,103 @@ func main() {
 		Zone:           *zone,
 		ServerPoolName: *serverPoolName,
 	}).SetupWithManager(mgr); err != nil {
-		klog.Error(err, "Error setting up EndpointSliceReconciler")
+		klog.Fatalf("Error setting up EndpointSliceReconciler: %v", err)
 	}
 
-	errChan := make(chan error)
+	// Start the controller manager
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
+
 		if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
-			klog.Error(err, "Error running manager")
-			errChan <- err
+			errChan <- fmt.Errorf("controller manager failed to start: %w", err)
 		}
+		klog.Info("controller manager started")
```

Comment:
At this point the serve is actually exiting not starting, can we please correct the log line?